File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    const string mergeAlternately(const string word1, const string word2) {
        int i = 0;
        string result;
        while (i < word1.size() || i < word2.size()) {
            if (i < word1.size()) {
                result += word1[i];
            }
            if (i < word2.size()) {
                result += word2[i];
            }
            ++i;
        }
        return result;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        output = ""
        length = len(word1) if len(word1) <= len(word2) else len(word2)

        for i in range(0, length+1):
            if i == len(word1) and i == len(word2):
                return output
            elif i == len(word1):
                return output + word2[i:]
            elif i == len(word2):
                return output + word1[i:]
            output += word1[i]
            output += word2[i]